src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -232,10 +232,6 @@ private function mapRequestPayload(Request $request, ArgumentMetadata $argument,
 
     private function mapUploadedFile(Request $request, ArgumentMetadata $argument, MapUploadedFile $attribute): UploadedFile|array|null
     {
-        if (!($files = $request->files->get($attribute->name ?? $argument->getName(), [])) && ($argument->isNullable() || $argument->hasDefaultValue())) {
-            return null;
-        }
-
-        return $files;
+        return $request->files->get($attribute->name ?? $argument->getName());
```

Comment:
What looks strange to me in the previous code is defaulting to an array without caring about the accepted type. To reduce the behavioral diff, we should continue to default to an array when arrays are accepted - and when not, then we should fix the behavior.

This would need a new test case for the array-case.

```suggestion
        if (!($files = $request->files->get($attribute->name ?? $argument->getName())) && ($argument->isNullable() || $argument->hasDefaultValue())) {
            return null;
        }

        return $files ?? ('array' === $argument->getType() ? [] : null);
```

Reply:
I see at least 4 test cases that should be verified when no file is provided:
* `#[MapUploadedFile] array $files` => should return an empty array
* `#[MapUploadedFile] ?array $files` => should return `null`
* `#[MapUploadedFile] UploadedFile $file` => should return `null` and an exception will be thrown
* `#[MapUploadedFile] UploadedFile ?$file` => should return `null`

Reply:
I understand returning an empty array for the sake of BC, but IMO an absence of value should be denoted by `null` and not an empty array. The user indicates the value is optional by making the argument nullable (that's also what the docs currently say), otherwise an `HttpException` should be thrown.

Reply:
That's not how this works for arrays for the other value resolvers. We have to be consistent.

Reply:
Alright, I applied your suggestion and added the missing test cases. I am not sure if the failed pipelines are related to my changes.